sweep/tx_input_set.go

```diff
@@ -71,6 +74,20 @@ func (b *txInputSet) add(input txInput) bool {
 		return false
 	}
 
+	// If this input comes from the wallet, verify that we still gain
+	// something with this transaction.
+	_, fromWallet := input.(*walletTxInput)
+	if fromWallet {
+		newWalletTotal := b.walletInputTotal + value
```

Comment:
taken in isolation, it looks like this can make the effective yield go down. Zooming out it should be okay, since we only add this when we are below the dustlimit, which obviously is what we want, even though it effectively costs us more because of the extra input. But this is not obvious from reading it in isolation.

Some comments would go a long way making it more clear I think.

Reply:
Comments added

Reply:
Nice, easier to read now :)